### PR TITLE
cli: initial implementation

### PR DESCRIPTION
**Pull request by jirikuncar**
Signed-off-by: Jiri Kuncar <jiri.kuncar@cern.ch>

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -43,15 +45,17 @@
...
     app = Flask('testapp')
-    FlaskMenu(app)
-    FlaskCLI(app)
-    breadcrumbs.Breadcrumbs(app=app)
-    InvenioDB(app)
     app.config.update(
         TESTING=True,
         SQLALCHEMY_DATABASE_URI=os.getenv('SQLALCHEMY_DATABASE_URI',
-                                          'sqlite://'),
+                                          'sqlite:///test.db'),
```
Comment (hachreak) [inline tests/conftest.py]: why don't you save in memory the db?

File: examples/app.py [hunk 2]
```diff
@@ -48,17 +48,13 @@
 
 # Create Flask application
```
Comment (hachreak) [inline examples/app.py]: can you add a minimum documentation to quick start as e.g. in oauthclient?

https://github.com/inveniosoftware/invenio-oauthclient/blob/master/examples/app.py#L28-L42

File: invenio_collections/cli.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+from flask_cli import with_appcontext
+from invenio_db import db
+
+
+#
+# Collection management commands
+#
+
+@click.group()
+def collections():
```
Comment (hachreak) [inline invenio_collections/cli.py]: can you add a command to show the collections tree?

File: invenio_collections/models.py [hunk 4]
```diff
@@ -37,10 +39,52 @@ def __repr__(self):
                 'dbquery: {0.dbquery}>'.format(self))
 
     id = db.Column(db.Integer, primary_key=True)
+
     name = db.Column(db.String(255), unique=True, index=True, nullable=False)
 
     dbquery = db.Column(db.Text, nullable=True)
 
+    @staticmethod
```
Comment (jirikuncar) [inline invenio_collections/models.py]: classmethod

File: invenio_collections/models.py [hunk 5]
```diff
@@ -37,10 +39,52 @@ def __repr__(self):
...
+            except NoResultFound:
+                # parent doesn't exists
+                raise CollectionError(
+                    "No collection found with name {}".format(parent_name))
+
+        db.session.add(collection)
+        return collection
+
+    @staticmethod
+    def attach(names, parent_name):
```
Comment (jirikuncar) [inline invenio_collections/models.py]: move back to cli as there is already `move_inside` method

File: examples/app.py [hunk 6]
```diff
@@ -38,8 +47,8 @@
 import os
 
 from flask import Flask
-from flask.ext import breadcrumbs
 from flask_babelex import Babel
+from flask_breadcrumbs import Breadcrumbs as FlaskBreadcrumbs
```
Comment (lnielsen) [inline examples/app.py]: Seems like ``from flask_breadcrumbs import Breadcrumbs`` would also work fine?

File: invenio_collections/errors.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+"""Invenio module for organizing metadata into collections."""
+
+from __future__ import absolute_import
+
+
+class CollectionError(Exception):
+    """Error generated when try to manage collections."""
+
+    pass
```
Comment (lnielsen) [inline invenio_collections/errors.py]: Not needed I think

File: invenio_collections/cli.py [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Click command-line interface for collection management."""
+
+from __future__ import absolute_import, print_function
+
+from functools import wraps
+
+import click
+from asciitree import LeftAligned
```
Comment (lnielsen) [inline invenio_collections/cli.py]: Nice, didn't know such a package existed :-)

**Comment by lnielsen** [comment]
LGTM - just have a look at the tiny cosmetics.

File: invenio_collections/errors.py (hunk 7 above)
Comment (jirikuncar) [inline invenio_collections/errors.py]: I had pep257 issue - missing line after class docstring - but I can't keep empty line at the end of file

**Pull request merged**